blackbird_python/blackbird/tests/test_auxiliary.py

```diff
@@ -181,6 +181,14 @@ def test_function_exp(self, parser, ctx, n, expected, num):
         expression = num(n)
         assert _func(func, expression) == np.exp(expected)
 
+    @pytest.mark.parametrize('n, expected', test_complex)
+    def test_function_log(self, parser, ctx, n, expected, num):
+        """Test that a Blackbird log function is properly called"""
+        func = blackbirdParser.FunctionContext(parser, ctx)
+        func.LOG = lambda: True

```

Comment:
just curious, what is the role of these lambdas?

Reply:
The way the parser works, if it detects a function, it produces a `FunctionContext` object with various method, each method corresponding to the function type.

So if `log(0.54)` is in the Blackbird code, the parser creates a `func = FunctionContext()` object with `func.LOG()` returning `True`.

Reply:
:thinking: